### DISCUSSION TITLE
;undefined\temp\puppeteer_dev_chrome_profile-XXXXXX&#039; please help to fix the error

### DISCUSSION THREAD

**Discussion by vpozniak** [spatie/browsershot#772] [Bugs]
Output:
================

Error Output:
================
[Error: ENOENT: no such file or directory, mkdtemp &#039;undefined\temp\puppeteer_dev_chrome_profile-XXXXXX&#039;] {
  errno: -4058,
  code: &#039;ENOENT&#039;,
  syscall: &#039;mkdtemp&#039;,
  path: &#039;undefined\\temp\\puppeteer_dev_chrome_profile-XXXXXX&#039;
}
 in file C:\vitalikpozniak\laragon\www\downkount-api\vendor\spatie\browsershot\src\Browsershot.php on line 955

**Comment by vpozniak** [comment]
$screenshotData = Browsershot::url($url)
            ->newHeadless()
            ->setNodeBinary('C:\Program Files\nodejs\node.exe')
            ->setNpmBinary('C:\Program Files\nodejs\npm')
            ->fullPage()
            ->setScreenshotType('png')
            ->screenshot();
        file_put_contents(storage_path('screenshots\screenshot.png'), $screenshotData);

**Reply by johnwaweru** [reply]
Did you manage to solve this problem?

**Comment by dguzman1** [comment]
I have the same problem in a project:

Laravel 10.46.0v
NodeJs 18.16.0v
PHP 8.2.12v

try reinstall all dependencies but, still not working :/

web.php
-----------
Route::get('/test-pdf', [TestPdfController::class,'printPdf']);

Controller
------------
namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Spatie\Browsershot\Browsershot;
use Illuminate\Support\Facades\File;

class TestPdfController extends Controller
{
    public function printPdf()
    {
	$path = public_path("/pdf_temp/");

        if (!File::exists($path)) {
            File::makeDirectory($path, 0755, true);
        }

    	Browsershot::url('https://google.com')
        ->newHeadless()
        ->setNodeBinary('C:\Users\pc\AppData\Roaming\nvm\v18.16.0\node')
        ->setNpmBinary('C:\Users\pc\AppData\Roaming\nvm\v18.16.0\npm')
        ->setNodeModulePath('C:\Users\pc\AppData\Roaming\nvm\v18.16.0\node_modules')
        ->noSandbox()
        ->format('A4')
        ->showBackground()
        ->save($path.'urlToPdf.pdf');
        echo 'Pdf generated at <i>public/pdf_temp/urlToPdf.pdf</i>';
    }
}

Error
-----------------------
The command "C:\Users\pc\AppData\Roaming\nvm\v18.16.0\node "C:\Users\pc\www\...\vendor\spatie\browsershot\src/../bin/browser.cjs" "{""url"":""https:\/\/google.com"",""action"":""pdf"",""options"":{""path"":""C:\\Users\\pc\\www\\...\\public\\\/pdf_temp\/urlToPdf.pdf"",""args"":[""--no-sandbox""],""viewport"":{""width"":800,""height"":600},""newHeadless"":true,""format"":""A4"",""printBackground"":true}}"" failed. Exit Code: 1(General error) Working directory: C:\Users\pc\www\...\public Output: ================ Error Output: ================ [Error: ENOENT: no such file or directory, mkdtemp 'undefined\temp\puppeteer_dev_chrome_profile-XXXXXX'] { errno: -4058, code: 'ENOENT', syscall: 'mkdtemp', path: 'undefined\\temp\\puppeteer_dev_chrome_profile-XXXXXX' }

I need to use browsershot since I must use tailwind for the all pdf

some help please!
Regards!

**Reply by tsitsi98** [reply]
@dguzman1 did you manage to solve this problem?

**Reply by dguzman1** [reply]
Unfortunately not, I decided to use **barryvdh/laravel-dompdf**

**Reply by tsitsi98** [reply]
@dguzman1 can you use tailwind using this barryvdh?

**Reply by dguzman1** [reply]
I couldn't, I had to create a specific style sheet for the pdfs, following the dependency limitations

**Reply by johnwaweru** [reply]
> @dguzman1 did you manage to solve this problem?

Here's the setup that worked for me, along with the steps I took:

```
use Spatie\Browsershot\Browsershot;
use Spatie\LaravelPdf\Facades\Pdf;


Pdf::view('index', ['data' => $data])
          ->format('a4')
          ->withBrowsershot(function (Browsershot $browsershot) {
            $browsershot->setNodeBinary('/home/ubuntu/.nvm/versions/node/v20.11.1/bin/node')
              ->setNpmBinary('/home/ubuntu/.nvm/versions/node/v20.11.1/bin/npm')
              ->setChromePath("/home/ubuntu/.nvm/versions/node/v20.11.1/lib/node_modules/puppeteer/.local-chromium/linux-1036745/chrome-linux/chrome")
              ->waitUntilNetworkIdle()
              ->timeout(120);
          })
          ->save($pdfName);
```
      
I also had to adjust the permissions within ~/home/ubuntu.

**IMPORTANT: Before modifying any permissions, ensure you do not alter those of the .ssh directory, as doing so could lock you out of your VPS. The current permissions should be:**

`drwx------ 2 ubuntu ubuntu  4096 Mar 19 17:59 .ssh`

Your default could be different. Do not change it from that.

To get things working, I initially made everything under /home/ubuntu executable by everyone. This approach is not recommended due to security concerns, but it served as a temporary fix. You can then gradually tighten the permissions as needed until you encounter issues again.

Lastly, ensure Chrome is installed globally and locate its exact path, typically found within the global node_modules directory. Ensure that this is also executable by everyone.

Ultimately, the most issues I encountered with browsershot had to do with permissions.

**Reply by dguzman1** [reply]
Let me try and I will answer you!!!
